python/llm/src/bigdl/llm/transformers/model.py

```diff
@@ -210,6 +210,20 @@ def from_pretrained(cls,
                 kwargs["torch_dtype"] = torch.float32
             optimize_model = False
             kwargs["modules_to_not_convert"] = ["lm_head"]
+        else:
+            from .llm_patching import is_bigdl_patched, patched_training_mode
+            if is_bigdl_patched:
+                # To handle finetuning patch of `load_in_8bit=True` etc
+                if kwargs.get('load_in_8bit', False):
+                    kwargs.pop('load_in_8bit')
+                if patched_training_mode == "lora":
+                    load_in_low_bit = "bf16"
+                elif patched_training_mode == "qalora":
+                    load_in_low_bit = "sym_int4"
+                elif patched_training_mode == "qlora" or patched_training_mode == "relora":
+                    load_in_low_bit = "nf4"
```

Comment:
Not needed - only need to support QLoRA or LoRA (based on BnB config)

Reply:
> Not needed - only need to support QLoRA or LoRA (based on BnB config)

I think LoRA fails to use BnB config. Current implementation of considering `patched_training_mode` would better handle LoraConfig setting and model convert.

For example, according to my local experiment, if I add `llm_patch(training_mode='lora')` at the beginning of https://github.com/tloen/alpaca-lora/blob/main/finetune.py, lora finetuning is in progress on GPU. And if I add `llm_patch(training_mode='qlora')` at the beginning, qlora finetuning will be in progress.

Reply:
> > Not needed - only need to support QLoRA or LoRA (based on BnB config)
> 
> I think LoRA fails to use BnB config. Current implementation of considering `patched_training_mode` would better handle LoraConfig setting and model convert.
> 
> For example, according to my local experiment, if I add `llm_patch(training_mode='lora')` at the beginning of https://github.com/tloen/alpaca-lora/blob/main/finetune.py, lora finetuning is in progress on GPU. And if I add `llm_patch(training_mode='qlora')` at the beginning, qlora finetuning will be in progress.

I think you may use various configs (BnB, load-in-4-bit, etc.) to choose LoRA vs. QLoRA

Reply:
> > I think you may use various configs (BnB, load-in-4-bit, etc.) to choose LoRA vs. QLoRA
> 
> OK, I'll modify the conditions of choosing training mode. BTW, we only need to consider patch case of LoRA and QLoRA?

Yes - existing peft code does not have qa-lora or relora